src/pages/api/donate/create/[id].ts

```diff
@@ -0,0 +1,57 @@
+import { PrismaClientKnownRequestError } from "@prisma/client/runtime";
+import { NextApiHandler, NextApiRequest, NextApiResponse } from "next";
+import { createBlankDonation } from "~/lib/db/donation";
+import { withSessionRoute } from "~/lib/withSession";
+
+const handler: NextApiHandler = async (req, res) => {
+  switch (req.method) {
+    case "POST":
+      handlePost(req, res);
+      break;
+    default:
+      res.setHeader("Allow", ["POST"]);
+      res.status(405).end(`Method ${req.method} Not Allowed`);
+  }
+};
+
+const handlePost = async (
+  request: NextApiRequest,
+  response: NextApiResponse
+) => {
+  const user = request.session.user;
+
+  if (!user || user.isLoggedIn === false) {
+    response.status(401).json({
+      message: "Not Authorized",
+    });
+    return;
+  }
+
+  const { id } = request.query;
+  console.log(request.body);
+  const { subaddress } = request.body?.data;
+
+  if (!id) throw new Error("Provide a streamer id");
```

Comment:
We need to find a consistent way for errors in http handlers. Somehow application and user facing errors should be combined.